nau_openedx_extensions/filters/pipeline.py

```diff
@@ -77,3 +80,51 @@ def _is_user_email_allowed(user, domains_allowed):
             if user_domain == domain or fnmatch(user_domain, f"*.{domain}"):
                 return True
         return False
+
+
+class FilterUsersWithAllowedNewsletter(PipelineStep):
+    """
+    Filter users with allowed newsletter.
+
+    Example usage:
+
+    Add the following configurations to your configuration file:
+
+    ```
+    OPEN_EDX_FILTERS_CONFIG = {
+        "org.openedx.learning.schedule.queryset.requested.v1": {
+            "fail_silently": False,
+            "pipeline": [
+                "nau_openedx_extensions.filters.pipeline.FilterUsersWithAllowedNewsletter",
+            ],
+        },
+    }
+    ```
+    """
+
+    def run_filter(self, schedules: QuerySet) -> dict:  # pylint: disable=arguments-differ
+        """
+        Execute filter that filters users with allowed newsletter.
+
+        Arguments:
+            schedules (QuerySet): Queryset of schedules to be sent.
+
+        Raises:
+            PreventScheduleQuerysetRequest: If the filter can't be applied.
+
+        Returns:
+            dict: Dictionary with the filtered schedules.
+        """
+        try:
+            filtered_schedules = schedules.filter(
+                enrollment__user__nauuserextendedmodel__allow_newsletter=True,
+            )
+            return {
+                "schedules": filtered_schedules,
+            }
+        except FieldError as exc:
+            raise ScheduleQuerySetRequested.PreventScheduleQuerySetRequest(
+                "The filter can't be applied because the user model "
+                "doesn't have the attribute 'nauuserextendedmodel'.",
```

Comment:
In which case do we not have the `nauuserextendedmodel`? when no one has `nauuserextendedmodel`?

Reply:
No, only when the model does not exists.